tests/containers/podmansh.pm

```diff
@@ -0,0 +1,200 @@
+#   * container is launched with existing user id
+#   * container is launched with keep-id of the user who run the container
+# - Restore /etc/zypp/credentials.d/ credentials
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use Mojo::Base 'containers::basetest';
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use Utils::Systemd qw(systemctl);
+use utils;
+use containers::common;
+
+my $quadlet_dir = '/etc/containers/systemd';
+
+my $src_image = "registry.opensuse.org/opensuse/leap:15.6";
+my $quadlet_container = <<_EOF_;
+[Unit]
+Description=The podmansh container
+After=local-fs.target
+
+[Container]
+Image=$src_image
+ContainerName=podmansh
+UserNS=keep-id
+RunInit=yes
+DropCapability=all
+NoNewPrivileges=true
+
+# change homedir to match 'user-console' needle
+Environment=HOME=/home
+WorkingDir=/home
+
+# OpenQA serial terminal access
+AddDevice=/dev/ttyS0
+
+Exec=sleep infinity
+
+[Quadlet]
+# avoid infinite waiting for network by podman-user-wait-network-online.service
+# https://github.com/systemd/systemd/issues/28762
+DefaultDependencies=false
+
+[Install]
+RequiredBy=default.target
+_EOF_
+
+my $unit_name = 'podmansh.container';
+my $user;
+my $uid;
+my $systemd_user_path;
+my $initial_user_shell;
+
+sub run {
+    my ($self) = @_;
+    select_serial_terminal;
+
+    my $podman = $self->containers_factory('podman');
+
+    # prepare user for rootless containers
+    prepare_rootless($self);
+
+    install_packages('podmansh');
+
+    # check quadlet works
+    my $quadlet = '/usr/libexec/podman/quadlet';
+    assert_script_run("$quadlet -version");
+
+    # create user container
+    assert_script_run("mkdir -p $systemd_user_path");
+    assert_script_run(qq(printf "$quadlet_container" >"$systemd_user_path/$unit_name"));
+    record_info('Unit', script_output("sudo -su $username $quadlet -user -v -dryrun"));
+
+    # enable user linger
+    assert_script_run("loginctl enable-linger $uid");
+
+    $initial_user_shell = script_output(qq(getent passwd "$user" | cut -d':' -f7));
+    assert_script_run(qq(usermod -s /usr/bin/podmansh "$user"));
+
+    # prepare ssh key for passwordless login
+    my $sshkeypath = "/root/.ssh/id_ed25519";
+    if (script_run(qq(test -e "$sshkeypath.pub")) != 0) {
+        assert_script_run(qq(ssh-keygen -t ed25519 -f "$sshkeypath" -q -N ""));
+    }
+    my $usersshpath = "/home/$user/.ssh";
+    assert_script_run(qq(mkdir -p "$usersshpath" && chown "$user:$user" "$usersshpath"));
+    assert_script_run(qq(tee -a "$usersshpath/authorized_keys" <"$sshkeypath.pub"));
+
+    # pre-pull the image and make sure the service is running
+    script_retry("sudo -su $username sh -c 'cd; podman pull $src_image'", retry => 3, delay => 60, timeout => 180);
+    systemctl("--machine=$username@.host --user start podmansh");
+    systemctl("--machine=$username@.host --user is-active podmansh");
+
+    # change the dafault podmansh non-root shell to match 'text-logged-in-bernhard' needle
```

Comment:
Why is this needed?

Reply:
Thanks for the suggestion. Serial terminal sounds better than using graphical console. However, this solution still requires these workarounds, because there are static regexes in place to check for the valid prompt.